src/prefect/client/client.py

```diff
@@ -934,16 +940,21 @@ def create_project(self, project_name: str, project_description: str = None) ->
             }
         }
 
-        res = self.graphql(
-            project_mutation,
-            variables=dict(
-                input=dict(
-                    name=project_name,
-                    description=project_description,
-                    tenant_id=self.active_tenant_id,
-                )
-            ),
-        )  # type: Any
+        try:
+            res = self.graphql(
+                project_mutation,
+                variables=dict(
+                    input=dict(
+                        name=project_name,
+                        description=project_description,
+                        tenant_id=self.active_tenant_id,
+                    )
+                ),
+            )  # type: Any
+        except ClientError as exc:
+            if skip_if_exists and "'Uniqueness violation.'" in str(exc):
```

Comment:
I think instead of catching this on error we should instead make a query for a matching project with that name and if it exists then return there instead of having to rely on a substring in the exception raised. Something like:

```python
if skip_if_exists:
  query_project = {
            "query": {
                with_args("project", {"where": {"name": {"_eq": project_name}}}): {
                    "id": True
                }
            }
        }
   project = self.graphql(query_project).data.project
   if project:
       return None
```

Reply:
That leaves a possibility for a race condition if multiple commands happen to be running concurrently - while parsing the error message is unfortunate I think the existing logic is less error prone.

Reply:
👍